src/main/java/jenkins/plugins/git/GitStep.java

```diff
@@ -97,7 +98,7 @@ public ListBoxModel doFillCredentialsIdItems(@AncestorInPath Item project,
             return delegate.doFillCredentialsIdItems(project, url);
         }
 
-        public FormValidation doCheckUrl(@AncestorInPath Item project,
+        public FormValidation doCheckUrl(@AncestorInPath Job project,
```

Comment:
Agreed with KostyaSha (commenting above) -- I think better to just do an instanceof check in the parent.